zerver/lib/subdomains.py

```diff
@@ -76,4 +77,9 @@ def is_static_or_current_realm_url(url: str, realm: Realm) -> bool:
     if split_url.netloc == "" and split_url.scheme == "":
         return True
 
+    # S3 storage we control, if used, is also static and thus exempt
+    if settings.LOCAL_UPLOADS_DIR is None:
+        if url.startswith(get_public_upload_root_url()):
```

Comment:
I think `startswith` is sufficiently safe here because `get_public_upload_root_url()` ends with a `/`?

Reply:
Yeah, I think that's right; we should just make sure detail that reasoning in a comment if we're doing that.

Reply:
I think that's correct, but maybe we should add an assert statement of some sort?